File: examples/load-data.ts
/**
 * Example: Loading and querying sports motion data in TypeScript
 *
 * Run: npx tsx examples/load-data.ts
 */

import { readFileSync } from 'fs';
import { join } from 'path';

interface Sport {
  sportId: string;
  name: string;
  category: string;
  movementPattern: string;
  applicableGrades: string[];
  equipment: string[];
  venue: string;
  skillUnits: any[];
  teachingProgression: any[];
  assessmentRules?: any[];
}

function loadSport(sportId: string): Sport {
  const filePath = join(__dirname, '..', 'data', `${sportId}.json`);
  const raw = readFileSync(filePath, 'utf-8');
  return JSON.parse(raw);
}

// Load 50m sprint data
const sprint = loadSport('50m-sprint');

console.log(`🏃 Sport: ${sprint.name}`);
console.log(`   Category: ${sprint.category}`);
console.log(`   Movement Pattern: ${sprint.movementPattern}`);
console.log(`   Applicable Grades: ${sprint.applicableGrades.join(', ')}`);
console.log(`   Skill Units: ${sprint.skillUnits.length}`);
console.log(`   Equipment: ${sprint.equipment.join(', ')}`);

// Query all critical errors across all skill units
const criticalErrors = sprint.skillUnits.flatMap((unit: any) =>
  unit.commonErrors
    .filter((e: any) => e.severity === 'critical')
    .map((e: any) => ({
      phase: unit.name,
      error: e.name,
      visualCue: e.visualCue,
    })),
);

console.log(`\n🚨 Critical Errors (${criticalErrors.length}):`);
for (const err of criticalErrors) {
  console.log(`   [${err.phase}] ${err.error}`);
  console.log(`   → ${err.visualCue}`);
}

// Get teaching progression
console.log(`\n📚 Teaching Progression (${sprint.teachingProgression.length} phases):`);
for (const phase of sprint.teachingProgression) {
  console.log(`   Phase ${phase.order}: ${phase.name}`);
  console.log(`   → ${phase.goal}`);
}

// Query joint angle specifications for the acceleration phase
console.log(`\n📐 Joint Angle Specs (Acceleration Phase):`);
const accel = sprint.skillUnits.find((u: any) => u.name === '加速跑');
if (accel) {
  for (const [joint, spec] of Object.entries(accel.keyJoints) as any) {
    console.log(`   ${joint}: ${spec.angleRange[0]}°–${spec.angleRange[1]}°`);
    console.log(`   → ${spec.angleDescription}`);
  }
}

// Assessment rules
if (sprint.assessmentRules) {
  console.log(`\n📋 Assessment Rules (${sprint.assessmentRules.length}):`);
  for (const rule of sprint.assessmentRules) {
    console.log(`   Rule: ${rule.ruleId}`);
    console.log(`   → ${rule.fullScoreStandard.description}`);
    console.log(`   → Deduction Points: ${rule.deductionPoints.length}`);
    console.log(`   → Correction Drills: ${rule.correctionStrategy.drills.length}`);
  }
}

// Summary across all sports
console.log(`\n${'='.repeat(60)}`);
console.log('📊 Full Dataset Summary');
console.log('='.repeat(60));

const allSports = ['50m-sprint', 'standing-long-jump', '1min-rope-skipping', 'medicine-ball-throw', 'pull-up'];
let totalUnits = 0;
let totalErrors = 0;
let totalDrills = 0;
let totalRules = 0;

for (const sportId of allSports) {
  const sport = loadSport(sportId);
  totalUnits += sport.skillUnits.length;
  totalErrors += sport.skillUnits.reduce((sum: number, u: any) => sum + u.commonErrors.length, 0);
  totalDrills += sport.teachingProgression.reduce((sum: number, p: any) => sum + p.drills.length, 0);
  totalRules += sport.assessmentRules?.length || 0;
  console.log(`   ${sport.name}: ${sport.skillUnits.length} units, ${sport.skillUnits.reduce((s: number, u: any) => s + u.commonErrors.length, 0)} errors, ${sport.teachingProgression.reduce((s: number, p: any) => s + p.drills.length, 0)} drills`);
}

console.log(`\n   Total: ${totalUnits} skill units, ${totalErrors} documented errors, ${totalDrills} practice drills, ${totalRules} assessment rules`);
console.log(`   Sports: ${allSports.length}`);
console.log(`   Movement Patterns: linear-locomotion, vertical-jump, cyclic-coordination, overhead-throw, pull-strength`);

File: examples/load-data.py
"""
Example: Loading and querying sports motion data in Python

Run: python examples/load-data.py
"""

import json
from pathlib import Path

DATA_DIR = Path(__file__).parent.parent / "data"


def load_sport(sport_id: str) -> dict:
    file_path = DATA_DIR / f"{sport_id}.json"
    with open(file_path, "r", encoding="utf-8") as f:
        return json.load(f)


# Load 50m sprint data
sprint = load_sport("50m-sprint")

print(f"🏃 Sport: {sprint['name']}")
print(f"   Category: {sprint['category']}")
print(f"   Movement Pattern: {sprint['movementPattern']}")
print(f"   Skill Units: {len(sprint['skillUnits'])}")

# Query all critical errors across all skill units
print("\n🚨 Critical Errors:")
for unit in sprint["skillUnits"]:
    for error in unit["commonErrors"]:
        if error["severity"] == "critical":
            print(f"   [{unit['name']}] {error['name']}")
            print(f"   → {error['visualCue']}")

# Get teaching progression
print(f"\n📚 Teaching Progression:")
for phase in sprint["teachingProgression"]:
    print(f"   Phase {phase['order']}: {phase['name']} — {phase['goal']}")

# Assessment rules
if sprint.get("assessmentRules"):
    print(f"\n📋 Assessment Rules:")
    for rule in sprint["assessmentRules"]:
        print(f"   Rule: {rule['ruleId']}")
        print(f"   → {rule['fullScoreStandard']['description']}")
        print(f"   → Correction verbal cue: {rule['correctionStrategy']['verbalCue']}")

# Print summary stats across all sports
print(f"\n{'='*60}")
print("📊 Full Dataset Summary")
print("="*60)

all_sports = ["50m-sprint", "standing-long-jump", "1min-rope-skipping", "medicine-ball-throw", "pull-up"]
total_units = 0
total_errors = 0
total_drills = 0
total_rules = 0

for sport_id in all_sports:
    sport = load_sport(sport_id)
    units = len(sport["skillUnits"])
    errors = sum(len(u["commonErrors"]) for u in sport["skillUnits"])
    drills = sum(len(p["drills"]) for p in sport["teachingProgression"])
    rules = len(sport.get("assessmentRules", []))
    total_units += units
    total_errors += errors
    total_drills += drills
    total_rules += rules
    print(f"   {sport['name']}: {units} units, {errors} errors, {drills} drills, {rules} rules")

print(f"\n   Total: {total_units} skill units, {total_errors} documented errors, {total_drills} practice drills, {total_rules} assessment rules")
print(f"   Sports: {len(all_sports)}")
print(f"   Movement Patterns: linear-locomotion, vertical-jump, cyclic-coordination, overhead-throw, pull-strength")
